maven-core/src/main/java/org/apache/maven/lifecycle/internal/LifecycleModuleBuilder.java

```diff
@@ -64,13 +64,13 @@ public LifecycleModuleBuilder(
             MojoExecutor mojoExecutor,
             BuilderCommon builderCommon,
             ExecutionEventCatapult eventCatapult,
-            ProjectExecutionListener projectExecutionListener,
+            List<ProjectExecutionListener> listeners,
```

Comment:
Can you explain the purpose of this API change?

Reply:
Could you be more precise ? Which API change are you talking about ?  The change is mainly about switching to pure constructor injection and final fields whenever possible.

Reply:
AH, maybe the commits are misleading.  The 3 last commits (Fix xxx) should be merged into the "Use constructor injector" one, which would make things more clear.  Those are problems in the refactoring and should not be reviewed per se, they only fix the previous commit.  Let me fix that...